persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/ConversationDAOTest.kt

```diff
@@ -1015,6 +1015,46 @@ class ConversationDAOTest : BaseDatabaseTest() {
         assertNull(result.firstOrNull { it.id == conversation.id })
     }
 
+    @Test
+    fun givenOneOnOneConversations_whenGettingAllConversations_thenShouldReturnsOnlyActiveConversations() = runTest {
+        conversationDAO.insertConversation(conversationEntity1)
+        conversationDAO.insertConversation(conversationEntity2)
+
+        userDAO.insertUser(user1.copy(oneOnOneConversationId = conversationEntity1.id)) // user active one-on-one
+        userDAO.insertUser(user2.copy(oneOnOneConversationId = null)) // user without active one-on-one
+
+        memberDAO.insertMembersWithQualifiedId(listOf(member1, member2), conversationEntity1.id)
+        memberDAO.insertMembersWithQualifiedId(listOf(member1, member2), conversationEntity2.id)
+
+        conversationDAO.getAllConversationDetails().first().let {
+            assertEquals(1, it.size)
+            assertEquals(conversationEntity1.id, it.first().id)
+        }
+    }
+
+    @Test
+    fun givenOneOnOneConversationNotExisting_whenGettingOneOnOneConversationId_thenShouldReturnEmptyList() = runTest {
+        // given
+        userDAO.insertUser(user1.copy(oneOnOneConversationId = conversationEntity1.id))
+
+        // then
+        assertTrue(conversationDAO.getOneOnOneConversationIdsWithOtherUser(user1.id, protocol = ConversationEntity.Protocol.PROTEUS).isEmpty())
+    }
```

Comment:
👨🏼‍🍳🤌🏻 
I saw the changes in the query and I immediately thought of these tests.

Awesome.